Client/AccountSyncHandler.swift

```diff
@@ -0,0 +1,48 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+/// `AccountSyncHandler` exists to observe certain `TabEventLabel` notifications,
+/// and react accordingly.
+class AccountSyncHandler: TabEventHandler, Loggable {
+
+    let throttler = Throttler(seconds: 5.0, on: DispatchQueue.global(qos: .utility))
```

Comment:
This implementation of throttle isn't really throttling the call. This implementation doesn't force a call every 5 seconds if there is information to send. Because it cancels the task before it has a chance to fire it's possible if the user is very active, e.g. they trigger a new call every 4 seconds, the call could be postponed indefinitely.
Because of the nature of the task it's probably a non-issue tho as this will likely fire frequently enough for desired effect to be reached. Just something to keep in mind if QA start reporting bugs related to this and doing lots of things quickly.

Reply:
Yup I'll definitely note it as a comment in the ticket! Given how often this call can happen, I'd be okay with postponing until the user settles down on page loads within a tab.